### PR TITLE
Add Modal page for Design Manual

### PR DESCRIPTION
**Pull request by kurzn**
This is the updated modals page for the Design Manual, including updated content, layout, and guidelines. The recommendation was to launch this page in the DM even though the component is not yet built in Capital Framework. It will be added as a task to build to the CF backlog. 

Please review @stephanieosan @marteki @ascott1

### CODE REVIEW

**Comment by kurzn** [comment]
Can be viewed here: http://kurzn.github.io/design-manual/ui-toolkit/modals.html 

Will need to fix the navigation since the page is still using the old one.

**Comment by ascott1** [comment]
@stephanieosan & @marteki If you can view @kurzn's version and give it a :+1:, I can fix the conflicts and merge.

File: ui-toolkit/modals.md [hunk 1]
```diff
@@ -1,7 +1,148 @@
...
+
+<p>Modals can be effective for communicating a warning or alert outside the main web page. They can be triggered by a user interaction (like pressing a button to “save”) or without an interaction (like a timeout warning after a period of inactivity).</p>
+
+<h3 id="interruption">Interruption</h3>
+<p>Use modals to force the user to complete a task or make a decision that requires their full attention outside of the main workflow. A critical piece of this scenario is that you need the workflow to be interrupted to be effective. If not, explore other UI elements to accomplish the goal.</p> 
+
+<div class="content-50 content-first">
+  <h4 id="examples">Examples of when to use modals</h4>
+  <ul>
+    <li>Confirm an action that can’t be undone, such as “Are you sure you want to delete this?</li>
```
Comment (marteki) [inline ui-toolkit/modals.md:31]: Missing end quote

Comment (marteki) [inline ui-toolkit/modals.md:31]: Also, we'll probably want to code all the curly quotation marks as either `&ldquo;` for the opening one or `&rdquo;` for the closing ones.

File: ui-toolkit/modals.md [hunk 2]
```diff
@@ -1,7 +1,148 @@
...
+
+<p>Provide separate focus and hover states for the close minicon and any “next” action buttons.</p>
+
+<p>The find function (ctrl+F) will not search information contained within a modal window.</p> 
+
+<p>Include offscreen instructions that describe the modal dialog and how to interact with it.</p>
+
+<h4 id="inactive">Specific development directions</h4>
+  <ul>
+    <li>Modal windows should be marked with aria-hidden="true" and toggled to false when visible and given the role=dialog ARIA role.</li>
```
Comment (marteki) [inline ui-toolkit/modals.md:79]: The first three bullet points here are a bit confusing. Maybe break them apart or rewrite them, so it's clearer what is to be done for each piece when the modal is visible or not visible (if that's the deciding factor)?

Also, add little tick marks around all the `aria-hidden="true"` and `role="dialog"` attributes.

File: ui-toolkit/modals.md [hunk 3]
```diff
@@ -1,7 +1,148 @@
...
+    <li>When the modal dialog is displayed, the main content of the page should be marked with aria-hidden="true" to prevent screen readers from interacting with it.</li>
+    <li>Add role="alertdialog" to the modal window.</li>
```
Comment (marteki) [inline ui-toolkit/modals.md:81]: Clarify that `role="alertdialog"` is only to be used for errors, emergency warnings, and the like? https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alertdialog_role

File: ui-toolkit/modals.md (hunk 1 above)
Comment (Scotchester) [inline ui-toolkit/modals.md:31]: @marteki That's not necessary anymore.

Comment (Scotchester) [inline ui-toolkit/modals.md:31]: (If a website is set to use the UTF-8 charset.)

**Comment by jimmynotjim** [comment]
This has been sitting a long time. Anyone want to take on updating it so we can merge it and then start working on the code in Capital Framework?

**Comment by jimmynotjim** [comment]
Updated with fixes for @marteki's comments and merged. Live at https://cfpb.github.io/design-manual/page-components/modals.html

**Comment by nataliafitzgerald** [comment]
@kurzn @jimmynotjim 
Should this issue be closed? The modals page is live.

**Pull request merged**